docs/DeveloperGuide.md

```diff
@@ -36,3 +92,351 @@
 ## Instructions for manual testing
 
 {Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+
+
+## Real penalty shootout setting
+
+
+### Overview
+
+
+As we are making a penalty shootout game, we should make the procedure of a match in the game the same as that of a 
+real penalty shootout. Here is the procedure of a real penalty shootout:
+
+
+Teams take turns to kick from the penalty mark, until each has taken five kicks. However, if one side has scored more 
+goals than the other could possibly reach with all of its remaining kicks, the shoot-out immediately ends, regardless 
+of the number of kicks remaining; this basis is called "best-of-five kicks".
+
+
+If after five rounds of kicks, the teams have scored an equal number of goals (or neither team has scored any goals), 
+additional rounds of one kick each are used until one team scores and the other misses. This is known as sudden death.
+
+
+### Design
+
+
+The procedure is facilitated by `MatchStat`. It records current round number, match number, player score, Ai score, 
+whether it's the player's turn to shoot, whether the match ends, and whether the player wins. 
+
+The `updateStat(boolean isPlayer, boolean isGoal)` method converts the outcomes of player's commands into player score 
+and Ai score after `penalty` or `save` commands. 
+
+The `decideMatchEnd()` method decides whether a match ends based on the rules mentioned above. 
+
+The `updateForNewMatch()` method is called after a match ends, and the player starts a new match. The method resets all 
+the stats recorded except for the `matchCount`, which is added one to show that the player has finished one more match.
+
+As we now only need to record the match stats of one player, the class is implemented as a utility class. (i.e. Using 
+class level fields and methods.) We may change it to an instantiable class later when we need to record the match stats 
+of multiple players. Here is the class diagram of the class:
+
+
+![MatchStatClassDiagram.png](diagrams%2FMatchStatClassDiagram.png)
```

Comment:
Perhaps you could consider removing the "C" by adding the "hide" command so that it looks closer to the class diagrams covered in our lectures?